### PR TITLE
Drones now spawn with seasonal hats depending on active holidays [VERY Important PR]

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: Maintenance drones and cogscarabs can now spawn with holiday-related hats on some holidays!
/:cl:

Ask and you shall receive. Holidays can now have drone hats associated with them, and all drones will spawn with that holiday's hat if they don't have one already. If there are multiple holidays going on, it will pick randomly between the hats. Cogscarabs all spawn with the same hat in this event, though, since they're part of the same cult.

This is done code-wise by giving drone shells a new static list called `possible_seasonal_hats` which is built when the first drone shell initializes in`build_seasonal_hats()`. They also use a `seasonal_hats` variable set to TRUE by default, which disables the new function if it's false. Holiday datums now have a `drone_hat` variable that contains the typepath of a hat or mask that drones will spawn with during that holiday, which are currently as follows:
* **Labor Day:** Hard hat
* **Firefighter's Day:** Firefighter helmet
* **Doctor Day:** Nurse hat
* **Pirate's Day:** Pirate hat (duh!)
* **US Thanksgiving:** Top hat
* **Mayan Doomsday Anniversary:** Tribal mask
* **Christmas and festive season:** Santa hat
* **Easter:** Bunny ears

### CODE REVIEW

**Comment by Hatterhat** [comment]
yeah this is pretty critical, speedmerge it

**Comment by arsserpentarium** [comment]
I wish I'd figure out,how to put hats on electrical drones...

**Comment by coiax** [comment]
Make sure this doesn't conflict with the camogear of snowflake drones (such as the Bardrone).

**Comment by Xhuis** [comment]
Drones that spawn with hats already won't spawn with seasonal ones, so it's all good.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/clock_cult/clock_items/construct_chassis.dm [hunk 1]
```diff
@@ -101,3 +108,6 @@
 			F.uses_power = FALSE
 		for(var/obj/item/clockwork/slab/S in cached_stuff)
 			S.no_cost = TRUE
+		if(seasonal_hat && seasonal_hat != "none")
+			var/obj/item/hat = new seasonal_hat(construct)
+			construct.equip_to_slot_or_del(hat, slot_head)
```
[the remark this answers is not in the document]
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_items/construct_chassis.dm]: Cogscarabs don't spawn with hats normally, so it should be fine.

**Comment by MrMaclang** [comment]
speedmerge this very important pr that give more immurshun

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**